### DISCUSSION TITLE
source.on_metadata gives warning

### DISCUSSION THREAD

**Discussion by BH-NL** [savonet/liquidsoap#2856] [User Support]
Hi,

I was using liquidsoap V2.0.7 with windows.
To see and update metadata on a source i use "source.on_metadata", my complete script was working fine with no warnings.
Doing to the memory leak on version 2.0.7 (the combination 2.0.7 with ffmpeg) I took version 2.1.3.  In this version my script gives several warnings (all warnings are the same "warning 2" as below) . Probably because of the warnings, the source.on_metadata does not work anymore.

When i start my script I get the warning;

```
At liquidsoapconfig.liq, line 19, char 0-59:
source.on_metadata(s, fun (n) -> print("Got meta 1: #{n}"))

Warning 2: This source is unused, maybe it needs to
be connected to an output. 
```

Maybe I'm doing something wrong but I can't find out wat, the source is used and connected to an output. I simplified my script, to make things easier.
The simplified code;

` ######## BASIC SETTINGS 

settings.log.stdout.set(true)
settings.harbor.bind_addrs.set(["0.0.0.0"])
settings.harbor.verbose.set(true)
settings.log.level.set(4)
settings.tag.encodings.set(["UTF-8", "ISO-8859-1"])

settings.log.file.path.set("/liquidsoap-2.1.3-win64/log/liquidsoap.log")

####### INPUT S AND PRINT METADATA 

s = input.harbor("djstream",port=8003,password="hackme1",id="Live-DJ",icy=true)

source.on_metadata(s, fun (n) -> print("Got meta 1: #{n}")) 

######## OUTPUT 

#Output 320kbps
#Mp3 output to Icecast

output.icecast(
     %ffmpeg(format="mp3", %audio(codec="libmp3lame",samplerate=44100, channels=2, b="320k")),
     host = "localhost",
	 encoding = "ISO-8859-1",
     port = 8000,
     password = "hackme",
     mount = "/OUT-320kbps",
	 public = false,
	 icy_metadata = "true",
	 headers =[("User-Agent","Liquidsoap 2.1.3 (Windows version)")],
	 fallible=true,
     s
) `

Can anyone help me?

Regards

**Comment by Russsgithub** [comment]
@BH-NL 

I think if you replace the ```source.on-metadata``` line with

``` s.on_metadata(fun(n) -> print("Got meta 1: #{n}")) ```

It'd work.

**Reply by BH-NL** [reply]
@Russsgithub 

Thanks! This solved my problem! 🙂

**Comment by Russsgithub** [comment]
Great.No worries.On 23 Jan 2023 20:01, BH-NL ***@***.***> wrote:

@Russsgithub

Thanks! This solved my problem! 🙂
